app/marketing/views.py

```diff
@@ -723,6 +725,120 @@ def org_settings(request):
     return TemplateResponse(request, 'settings/organizations.html', context)
 
 
+def tax_settings(request):
+    """Display and save user's Tax settings.
+
+    Returns:
+        TemplateResponse: The user's Tax settings template response.
+
+    """
+    msg = ''
+    profile, es, user, is_logged_in = settings_helper_get_auth(request)
+
+    if not user or not profile or not is_logged_in:
+        login_redirect = redirect('/login/github?next=' + request.get_full_path())
+        return login_redirect
+        
+    # location  dict is not empty
+    location = ''
+    if profile.location:
+        location_components = json.loads(profile.location)
+        if 'locality' in location_components:
+            location += location_components['locality']
+        if 'country' in location_components:
+            country = location_components['country']
+            if location:
+                location += ', ' + country
+            else:
+                location += country
+        if 'code' in location_components:
+            code = location_components['code']
+            if location:
+                location += ', ' + code
+            else:
+                location += code
+    # location dict is empty
+    else:
+        # set it to the last location registered for the user
+        location_components = profile.locations[-1]
+        if 'city' in location_components:
+            location += location_components['city']
+        if 'country_name' in location_components:
+            country_name = location_components['country_name']
+            if location:
+                location += ', ' + country_name
+            else:
+                location += country_name
+        if 'country_code' in location_components:
+            country_code = location_components['country_code']
+            if location:
+                location += ', ' + country_code
+            else:
+                location += country_code
+    
+    #address is not empty
+    if profile.address:
+        address = profile.address   
+    else:
+        address = ''
+
+    if request.POST:
+
+        if 'preferred_payout_address' in request.POST.keys():
```

Comment:
do we need all of these other settings inside of the tax settings update method?

Reply:
Hi @owocki , I left them in this method because i see them in many other setting methods. I can remove this code part without problems.

Reply:
i think remove them. if its copy/pasted elsewhere then thats something that should be removed in another PR

Reply:
Done